internal/ir/xds.go

```diff
@@ -179,8 +179,8 @@ type HTTPListener struct {
 	// Refer to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#config-route-v3-virtualhost
 	// for more info.
 	Hostnames []string `json:"hostnames" yaml:"hostnames"`
-	// Tls certificate info. If omitted, the gateway will expose a plain text HTTP server.
-	TLS []*TLSListenerConfig `json:"tls,omitempty" yaml:"tls,omitempty"`
+	// Tls configuration. If omitted, the gateway will expose a plain text HTTP server.
+	TLS *TLSListenerConfig `json:"tls,omitempty" yaml:"tls,omitempty"`
```

Comment:
lets rename the struct to `TLSConfig` since its being reused here, and this isn't a listener, its a client

Reply:
Renamed.